src/content/app/genome-browser/state/browser-general/browserGeneralSlice.ts

```diff
@@ -265,8 +257,53 @@ const browserGeneralSlice = createSlice({
     ) {
       state.activeFocusObjectIds = action.payload;
     },
-    updateTrackStates(state, action: PayloadAction<BrowserTrackStates>) {
-      state.trackStates = merge(state.trackStates, action.payload);
+    updateObjectTrackStates(
+      state,
+      action: PayloadAction<{
+        status: Status;
+        transcripts?: string[];
+      }>
+    ) {
+      const activeGenomeId = state.activeGenomeId as string;
+      const activeFocusObjectId = state.activeFocusObjectIds[
+        activeGenomeId
+      ] as string;
+      const { status, transcripts } = action.payload;
+
+      state.trackStates = set(
+        `${activeGenomeId}.objectTracks.${activeFocusObjectId}.status`,
+        status,
+        state.trackStates
+      );
+
+      if (transcripts) {
+        state.trackStates = set(
+          `${activeGenomeId}.objectTracks.${activeFocusObjectId}.transcripts`,
+          transcripts,
+          state.trackStates
+        );
+      }
+
+      browserStorageService.saveTrackStates(state.trackStates);
```

Comment:
Here and below — redux team urges people not to run side effects in reducers.

Reply:
✅